sw/banshee/src/tran.rs

```diff
@@ -1194,6 +1199,30 @@ impl<'a> InstructionTranslator<'a> {
         Ok(())
     }
 
+    unsafe fn emit_imm12_rs1(&self, data: riscv::FormatImm12Rs1) -> Result<()> {
+        let imm = data.imm();
+        trace!("{} x{}, x{}", data.op, data.rs1, imm);
+
+        // Compute the address.
+        let rs1 = self.read_reg(data.rs1);
+        let imm = LLVMConstInt(LLVMInt32Type(), (imm as i64) as u64, 0);
+
+        // Perform the operation.
+        match data.op {
+            riscv::OpcodeImm12Rs1::Scfgwi => {
+                // ssr write immediate holds address offset in imm12, content in rs1
+                let addr = LLVMBuildAdd(
+                    self.builder,
+                    LLVMConstInt(LLVMInt32Type(), SSR_BASE, 0),
+                    imm,
+                    NONAME,
+                );
+                self.write_mem(addr, rs1, 2);
```

Comment:
That's a clever way to solve this for now! Later we'll probably want to directly call into some SSR config read/write functions here, and forward accesses inside `write_mem` to those functions as well (might already be halfway implement). But no need to do this now 👍

Reply:
Oh you mean using `banshee_ssr_write_cfg`?